vispy/visuals/transforms/default_transform.py

```diff
@@ -0,0 +1,394 @@
+                                     self._has_scaling or 
+                                     self._has_rotation)
+        else:
+            self._is_identity = False
+    
+    ## Methods that change the transformation
+    
+    def reset(self):
+        """ Set the transform back to the identity transform.
+        """
+        self._reset()
+        self.update()
+    
+    def set_matrix(self, matrix):
+        """ Set the transform to the given 4x4 matrix
+        
+        Techcnically, the given matrix can describe any transformation,
+        but it is recommended to restrict to transformations composed
+        only of scaling, rotations and translations.
+        
+        It is recommended to not use this method unless the transform
+        can only be described with a matrix, because setting the matrix
+        will put the transform in a mode where inversion and composition
+        are less efficient.
+        """
+        matrix = np.array(matrix, copy=False)  # if list or matrix was given
+        assert matrix.shape == (4, 4)
+        self._matrix = matrix
+        self.update()
+    
+    # todo: set_mapping 2D and 3D
+    
+    # todo: set_ortho? set_perspective?
+    
+    def translate(self, offset):
+        """ Translate the transform.
+        
+        The translation is applied *after* the transformations already
+        present in the matrix.
+        """
+        dx, dy, dz = offset
+        
+        if self._matrix is None:
+            tx, ty, tz = self._translation
+            self._translation = float(tx+dx), float(ty+dy), float(tz+dz)
+            self._has_translation = any(self._translation)
+        else:
+            transmat = transforms.translate((dx, dy, dz))
+            self._matrix = np.dot(self._matrix, transmat)
+        
+        self.update()
+    
+    def rotate(self, angle, axis=None, deg=True):
+        """ Rotate the transform with angle degrees/radians around the axis.
+        
+        Also accepts a Quaternion object as a single argument.
+        
+        The rotation is applied *after* the transformations already
+        present in the matrix.
+        """
+        if isinstance(angle, Quaternion):
+            q = angle
+        else:
+            angle = float(angle)
+            ax, ay, az = axis
+            q = Quaternion.create_from_axis_angle(angle, ax, ay, az, deg)
+            
+        if self._matrix is None:
+            tx, ty, tz = self._translation
+            self._translation = q.rotate_point((tx, ty, tz))
+            self._rotation = (q * self._rotation)#.normalize()
+            qq = self._rotation
+            self._has_rotation = qq.w != 1.0 or any(tuple(qq)[1:])
+        else:
+            rotmat = q.get_matrix()
+            self._matrix = np.dot(self._matrix, rotmat)
+        
+        self.update()
+    
+    def scale(self, scale):
+        """ Scale the transform with a scalar value or a 3-element tuple.
```

Comment:
It would be nice to have the option of changing the `_scaling` and `_rotation` parameters directly, rather than having to rebuild the entire transform.